Ability to override deserializer in Kafka consumer

ilayaperumalg opened the pull request:
- When binding the consumer, the kafka consumer should not be set to use `ByteArrayDeserializer` for both key/value deserializer. Instead, they need to be used as the default values. Any extended consumer properties for key/value deserializer should override this default deserializer.
- Add test

This resolves #55

mbogoevici commented on spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java:
```diff
@@ -16,9 +16,14 @@
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.junit.Assert.*;
```
Switch to individual imports.

mbogoevici replied:
Use individual imports.

mbogoevici commented on spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java:
```diff
@@ -1187,6 +1198,46 @@ public void testPartitionCountNotReduced() throws Exception {
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testConsumerCustomDeserializer() throws Exception {
```
It is not enough to test that overriding the properties works. We need to test that sending and receiving messages with these settings actually works.

mbogoevici commented on spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java:
```diff
@@ -303,27 +302,18 @@ protected MessageProducer createConsumerEndpoint(String name, String group, Coll
-		ConsumerFactory<byte[], byte[]> consumerFactory = new DefaultKafkaConsumerFactory<>(props, keyDecoder,
-				valueDecoder);
-
+		ConsumerFactory<byte[], byte[]> consumerFactory = new DefaultKafkaConsumerFactory<>(props);
```
We should also support customizing the serialization on the consumer side.

mbogoevici commented:
Tried to run this, but the branch runs into test failures constantly (master runs fine) - can you check @ilayaperumalg ?

ilayaperumalg commented:
@mbogoevici Initially I thought this was because of the inconsistent test behaviour before. Now I realized it was a mistake on my part not to unbind some of the bindings created during the tests. Fixed them and updated the PR.

mbogoevici commented:
Cool - thanks @ilayaperumalg!

mbogoevici commented:
Rebased, squashed and merged via https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/commit/36c90aa8c0acf1fe29e235b1d2aac05115fc4c90
